qiskit_aqua/algorithms/adaptive/qsvm/cost_helpers.py

```diff
@@ -82,6 +81,50 @@ def cost_estimate_sigmoid(shots, probs, gt_labels):
     Returns:
         float: averaged sigmoid cross entropy loss between estimated probs and gt_labels
     """
+    """
+    The prior version of this function errors out when the probabilities in other_probs have 
+    repeated values for most entries. For example, the bug can be simulated when 
+    the original version of cost_estimate_sigmoid is called with the below values for the 
+    function parameters.
+    probs=np.array([[0.14, 0.25, 0.45], [0.15, 0.25, 0.45], [0.14, 0.25, 0.14]])
+    gt_labels = np.array([0, 2, 1])
+
+    Specifically, the second line below errors out with a dimensionality error when
+    the setdiff1d call results in a single element array for other_probs,
+    and so other_probs[1] causes an error.
+    The logic for cross-entropy loss is not correct in this scenario. The calculation for 
+    cross-entropy loss has been fixed as part of this fork.
+    other_probs = np.setdiff1d(probs[other_index], rest_p)
+        numerator = np.sqrt(shots) * ((1. + np.abs(other_probs[0] - other_probs[1])) /
+                                      number_of_classes - rest_p)
+    """
+    epsilon = 1e-15
+    probs = np.clip(probs, epsilon, 1 - epsilon)
+    number_of_classes = probs.shape[1]
+
+    if number_of_classes > 1:
+        #if len(np.unique(gt_labels)) > number_of_classes:
+        #    raise ValueError('Too many classes compared to pred.')
+        y_true = np.eye(number_of_classes)[np.array(gt_labels).reshape(-1)]
+    else:
```

Comment:
do you think the else branch is necessary. It corresponds to "number_of_classes=1"

Reply:
The else branch refers to the scenario where it is a binary class classification problem (positive or negative for the single label). Since number_of_classes = probs.shape[1], this will be 1 when we have only a binary classification prob (where only 1 probability is required for each data point).
For the binary classification case, the cross entropy loss is simply -ylogp - (1-y)log(1-p) summed over all data points. That's basically why the else branch is there.